Number_types/include/CGAL/Number_types/internal/Exact_type_selector.h

```diff
@@ -93,8 +93,12 @@ struct Exact_ring_selector : Exact_field_selector < T > { };
 
 template <>
 struct Exact_ring_selector<double>
-#ifdef CGAL_HAS_MPZF
+#if BOOST_VERSION > 107900 && defined(CGAL_USE_BOOST_MP)
+{ typedef boost::multiprecision::cpp_int Type; };
+#elif CGAL_HAS_MPZF
 { typedef Mpzf Type; };
+ #elif defined(CGAL_USE_BOOST_MP)
+{ typedef boost::multiprecision::cpp_int Type; };
```

Comment:
The space before `#elif` should not be there (consistency.

Maybe should in the complete `#if ... #endif` construct all lines not starting wit `#` be indented (readability), although this is a bit "contrary" to normal `struct` definitions (see the definition a bit lower for `Exact_field_selector`) that start with `{` at the beginning of the line.